plaidml/bridge/openvino/ops/detection_output.cpp

```diff
@@ -0,0 +1,464 @@
+                                 op::minimum(cast(edsl::Tensor{1.0f}, DType::FLOAT32), decoded_bboxes));
+  }
+
+  return decoded_bboxes;
+}
+
+void registerDetectionOutput() {
+  registerOp("DetectionOutput", [](const Context& ctx) {
+    auto* layer = ngraph::as_type<ngraph::opset4::DetectionOutput>(ctx.layer);
+    IE_ASSERT(ctx.operands.size() == 3 || ctx.operands.size() == 5);
+    auto Location = ctx.operands.at(0);
+    auto Confidence = ctx.operands.at(1);
+    auto Priors = ctx.operands.at(2);
+
+    edsl::Tensor ArmConfidence;
+    edsl::Tensor ArmLocation;
+    bool with_add_pred = false;
+    if (ctx.operands.size() == 5) {
+      ArmConfidence = ctx.operands.at(3);
+      ArmLocation = ctx.operands.at(4);
+      with_add_pred = true;
+    }
+
+    auto num_classes = layer->get_attrs().num_classes;
+    auto background_label_id = layer->get_attrs().background_label_id;
+    auto top_k = layer->get_attrs().top_k;
+    auto variance_encoded_in_target = layer->get_attrs().variance_encoded_in_target;
+    auto keep_top_k = layer->get_attrs().keep_top_k;
+    auto code_type = layer->get_attrs().code_type;
+    auto share_location = layer->get_attrs().share_location;
+    auto nms_threshold = layer->get_attrs().nms_threshold;
+    auto confidence_threshold = layer->get_attrs().confidence_threshold;
+    auto clip_after_nms = layer->get_attrs().clip_after_nms;
+    auto clip_before_nms = layer->get_attrs().clip_before_nms;
+    auto decrease_label_id = layer->get_attrs().decrease_label_id;
+    auto normalized = layer->get_attrs().normalized;
+    auto input_height = layer->get_attrs().input_height;
+    auto input_width = layer->get_attrs().input_width;
+    auto objectness_score = layer->get_attrs().objectness_score;
+
+    int prior_size = normalized ? 4 : 5;
+    int prior_offset = normalized ? 0 : 1;
+    int i_h = normalized ? 1 : input_height;
+    int i_w = normalized ? 1 : input_width;
+    auto batch = Location.compute_shape().sizes()[0];
+    auto num_priors = Priors.compute_shape().sizes()[2] / prior_size;
+    auto priors_shape_variance = Priors.compute_shape().sizes()[1];
+
+    std::vector<int64_t> location_shape = {batch, num_priors, 4};
+    std::vector<int64_t> confidence_shape = {batch, num_priors, num_classes};
+    std::vector<int64_t> priors_shape = {batch, priors_shape_variance, num_priors, prior_size};
+
+    edsl::Tensor location = edsl::reshape(Location, location_shape);
+    edsl::Tensor confidence = edsl::reshape(Confidence, confidence_shape);
+    edsl::Tensor priors = edsl::reshape(Priors, priors_shape);
+
+    edsl::Tensor variances;
+    if (variance_encoded_in_target) {
+      // Fill variances with 1s.
+      priors =
+          op::slice(priors).add_dim(0, batch).add_dim(0, 1).add_dim(0, num_priors).add_dim(prior_offset, prior_size);
+      auto one = cast(edsl::Tensor{1}, priors.dtype());
+      auto var_shape = priors.compute_shape().sizes();
```

Comment:
I think you do not need to cast the shape to `vector<int>` I have changed `op::broadcast` to support int64 vectors in an earlier PR. Now you can use `variances = op::broadcast(one,var_shape, {});`

Reply:
Done.